### ISSUE TITLE
Alternate method for defining built-in or custom units

### ISSUE THREAD

**Issue by ericman314** [ericman314/UnitMath#11]
It would be terrific if custom and built-in units could be defined based off of other units.

Example: 

```js
const UNITS = {
  // length
  meter: {
    base: DIMENSIONS.LENGTH,
    prefixes: PREFIXES.LONG,
    commonPrefixes: ['nano', 'micro', 'milli', 'centi', '', 'kilo'],
    value: 1
  },
  m: {
    prefixes: PREFIXES.SHORT,
    commonPrefixes: ['n', 'u', 'm', 'c', '', 'k'],
    value: '1 meter'
  },
  inch: {
    value: '0.0254 m'
  },
  foot: '12 inch',
  ...
```

**Advantages:**
- Simpler way to define new units
- Retains backwards compatibility
- Default properties could be omitted (i.e., `PREFIXES.NONE`)
- Shorthand syntax (see `foot` versus `inch` above)
- Would reduce `UnitStore.js` file size
- Defining custom units and built-in units uses the exact same code
- Less prone to errors while defining units
- Wouldn't have to declare the unit's `DIMENSIONS` or determine the unit's value in SI units

**Disadvantages:**
- Would require a good deal of refactoring
- Longer processing time (only once per configuration, and config changes where only formatting options change wouldn't require re-processing)
- Would either have to 1) require that units are declared "in order", or 2) use multiple passes to process the units.
- Redefinitions of units such as `kg` would cause dependent units to change as well, which might be considered good or bad depending on the circumstances.

**Comment by ericman314** [comment]
The current work on this is in https://github.com/ericman314/UnitMath/tree/alternate-unit-defs.

Still need to:

- [ ] Clean up UnitStore.js
- [ ] Expose the definitions on the options
- [ ] Read additional definitions from the options
- [ ] Update the readme
- [ ] Write a whole lot of tests

**Comment by ericman314** [comment]
This has been released in v0.2.0. Hooray!
